src/core/symbology-ng/qgsgraduatedsymbolrendererv2.h

```diff
@@ -139,8 +140,8 @@ class CORE_EXPORT QgsGraduatedSymbolRendererV2 : public QgsFeatureRendererV2
 
     virtual QgsSymbolV2List symbols() override;
 
-    QString classAttribute() const { return mAttrName; }
-    void setClassAttribute( QString attr ) { mAttrName = attr; }
+    QString classAttribute() const { return mExpression.data() ? mExpression->dump() : ""; }
```

Comment:
I reverted to the former code so not needed anymore.

 I'm not completely sure of the temp string though. It may be that the compiler does RVO more easily with a QString(), but that would have to be checked, I agree on an assignment

    const QString s("hello")

is better than 

    const QString s = "hello"

but I believe in this case, returning a QString() or a QString("") (which is what the compiler does) is equivalent.